projects/ngx-charts-on-fhir/src/lib/fhir-mappers/simple-medication-mapper.service.ts

```diff
@@ -4,72 +4,53 @@ import { MedicationOrder } from 'fhir/r2';
 import { merge } from 'lodash-es';
 import { DataLayer } from '../data-layer/data-layer';
 import { Mapper } from '../fhir-converter/multi-mapper.service';
-import { ChartAnnotation } from '../utils';
-import { TIME_SCALE_OPTIONS, LINEAR_SCALE_OPTIONS, ANNOTATION_OPTIONS } from './fhir-mapper-options';
+import { TIME_SCALE_OPTIONS, CATEGORY_SCALE_OPTIONS } from './fhir-mapper-options';
 import { FhirMappersModule } from './fhir-mappers.module';
 
-/** Required properties for mapping an Observation with [SimpleMedicationMapper] */
+/** Required properties for mapping a MedicationOrder with [SimpleMedicationMapper] */
 export type SimpleMedication = {
   medicationCodeableConcept: {
     text: string;
   };
   dateWritten: string;
 } & MedicationOrder;
 export function isMedication(resource: MedicationOrder): resource is SimpleMedication {
-  return !!(
-    resource.resourceType === 'MedicationOrder' &&
-    resource.prescriber?.reference &&
-    resource.dosageInstruction &&
-    resource.medicationCodeableConcept
-  );
+  return !!(resource.resourceType === 'MedicationOrder' && resource.dateWritten && resource.medicationCodeableConcept?.text);
 }
 
-/** Maps a FHIR Observation resource that has a single valueQuantity */
+/** Maps a FHIR MedicationOrder resource that only has a dateWritten and no supply duration */
 @Injectable({
   providedIn: forwardRef(() => FhirMappersModule),
 })
 export class SimpleMedicationMapper implements Mapper<SimpleMedication> {
   constructor(
     @Inject(TIME_SCALE_OPTIONS) private timeScaleOptions: ScaleOptions<'time'>,
-    @Inject(LINEAR_SCALE_OPTIONS) private linearScaleOptions: ScaleOptions<'linear'>,
-    @Inject(ANNOTATION_OPTIONS) private annotationOptions: ChartAnnotation
+    @Inject(CATEGORY_SCALE_OPTIONS) private categoryScaleOptions: ScaleOptions<'category'>
   ) {}
   canMap = isMedication;
-//   created Medication Data Layer 
-  map(resource: MedicationOrder): any {
-    let endDate = new Date(resource.dateWritten ? resource.dateWritten : "" )
-    endDate.setDate(endDate.getDate() + 2) 
-    const output =  {
+  map(resource: SimpleMedication): DataLayer {
+    return {
       name: resource?.medicationCodeableConcept?.text,
       category: 'Medication',
       datasets: [
         {
           label: resource?.medicationCodeableConcept?.text,
-        //   yAxisID: resource.valueQuantity.unit,
+          yAxisID: 'medications',
+          indexAxis: 'y',
           data: [
-            [
-              new Date(resource.dateWritten ? resource.dateWritten : "" ),
-              endDate
-            ],
+            {
+              x: new Date(resource.dateWritten).getTime(),
+              y: resource?.medicationCodeableConcept?.text,
+            },
           ],
```

Comment:
we can now use a string as y-value because the new `TimelineDataPoint` type is default data type param for `DataLayer`